File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l = 0 
        r = 1
        maxi = 0 
        while r<len(prices):
            if prices[l] >= prices[r]:
                l = r
            elif prices[r] > prices[l]:
                maxi = max(maxi , prices[r] - prices[l])
            r += 1
        return maxi
     

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int l = 0;
        int r = 1; 
        int ma = 0;
        while (r<prices.size()){
            if (prices[r] < prices[l]) l = r;
            else if ( prices[l] <= prices[r]) ma = max(ma , prices[r]- prices[l]);
            r ++;
        }
        return ma;
    }
};